filebeat/prospector/prospector.go

```diff
@@ -52,16 +56,25 @@ func NewProspector(cfg *common.Config, states file.States, outlet Outlet) (*Pros
 		wg:            sync.WaitGroup{},
 		states:        &file.States{},
 		channelWg:     sync.WaitGroup{},
+		Once:          false,
 	}
 
-	if err := cfg.Unpack(&prospector.config); err != nil {
+	var err error
+	if err = cfg.Unpack(&prospector.config); err != nil {
 		return nil, err
 	}
-	if err := prospector.config.Validate(); err != nil {
+	if err = prospector.config.Validate(); err != nil {
 		return nil, err
 	}
 
-	err := prospector.Init(states)
+	var h map[string]interface{}
+	cfg.Unpack(&h)
+	prospector.Hash, err = hashstructure.Hash(h, nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I would really like to see a `.Hash()` method so we have one place to change the hashing logic if needed. I can get this done in a later PR when I plan to unify reloading in filebeat and metricbeat.